File: 2021/day_2/one.js
const coords = require('fs')
    .readFileSync('input', { encoding: 'utf-8' })
    .split('\n')
    .reduce(({ x, y }, value) => {
        let [way, qty] = value.split(' ')
        qty = parseInt(qty);
        if (way === 'forward') {
            return { x: x + qty, y };
        }
        if (way === 'up') {
            return { x, y: y - qty };
        }
        if (way === 'down') {
            return { x, y: y + qty };
        }
        return { x, y }
    }, { x: 0, y: 0 });
console.log(coords.x * coords.y);

File: 2021/day_2/one.py
x, y = 0, 0
for line in open('input').readlines():
    line = line.split()
    if line[0] == 'forward':
        x += int(line[1])
    if line[0] == 'up':
        y -= int(line[1])
    if line[0] == 'down':
        y += int(line[1])
print(x * y)